### ISSUE TITLE
No such file or directory: './model/search_codebert_switch/checkpoint-best-mrr/pytorch_model.bin'

### ISSUE THREAD

**Issue by skye95git** [Jun-jie-Huang/CoCLR#2]
Hi, I want to try the `CoCLR on Code Search`. When I run `Step 2: training and evaluating`, there is an error:

<img width="818" alt="捕获" src="https://user-images.githubusercontent.com/41561936/130894052-7bc6f09b-c191-42b3-9d1c-f5557bd37846.PNG">

What should I do? Do I need to create my own folder path `./model/search_codebert_switch`? Is the model `pytorch_model.bin` generated by itself, or does it need to be stored in the folder in advance?

**Comment by skye95git** [comment]
> I see. To solve this, you can first download the codesearchnet-checkpoint from [this link](https://drive.google.com/drive/folders/1rM5A6dPf05Q5mP9kWjfsdRIpsqfI4IBi?usp=sharing). Then move it to `./model/` and rename the dirname to `codesearchnet-checkpoint`

Hi, I have another question: I will train a search model with CoCLR in the step 3. Why do I need to download the checkpoint trained on CodeSearchNet in the step 1? What is the role of the checkpoint downloaded in step 1? Is it codebert as the encoder?

**Comment by Jun-jie-Huang** [comment]
> > I see. To solve this, you can first download the codesearchnet-checkpoint from [this link](https://drive.google.com/drive/folders/1rM5A6dPf05Q5mP9kWjfsdRIpsqfI4IBi?usp=sharing). Then move it to `./model/` and rename the dirname to `codesearchnet-checkpoint`
> 
> Hi, I have another question: I will train a search model with CoCLR in the step 3. Why do I need to download the checkpoint trained on CodeSearchNet in the step 1? What is the role of the checkpoint downloaded in step 1? Is it codebert as the encoder?

The reason why initializing with checkpoint trained on CodeSearchNet is to boost performance. We find the performance drops if  the checkpoint not incorporated. You can also remove line of command `--checkpoint_path ./model/codesearchnet-checkpoint \` to run.

**Comment by Jun-jie-Huang** [comment]
> > I see. To solve this, you can first download the codesearchnet-checkpoint from [this link](https://drive.google.com/drive/folders/1rM5A6dPf05Q5mP9kWjfsdRIpsqfI4IBi?usp=sharing). Then move it to `./model/` and rename the dirname to `codesearchnet-checkpoint`
> 
> Thanks for your reply! But the link need access:
> 
> <img alt="捕获" width="446" src="https://user-images.githubusercontent.com/41561936/131063822-ad680661-6882-4c36-999a-5fa0425985dd.PNG">

I see. I will update this link

**Comment by akhileshgotmare** [comment]
@Jun-jie-Huang you mention the checkpoint is trained on codesearchnet - is this training done using the L_b and L_ib losses from the paper or MLM loss of codebert?
